### PR TITLE
Add addtrap() helper to run-checks

### PR DESCRIPTION
**Pull request by mvo5**
We use a trap to display "endmsg" which tells about success or
failure of the tests. However the script also uses a second trap
to do cleanup which overrides the first trap. This means no success
or failure message at the end. This branch fixes this by allowing
stacking of trap functions.

### CODE REVIEW

**Comment by elopio** [comment]
I added one more trap:
trap "rm -f $TMP_INTEGRATION" 0

I didn't know that would override the others.

**Comment by mvo5** [comment]
@elopio thanks! I merged with upstream and add the "addtrap" for the new trap as well. And no worries, shell is a tricky language (despite what people say :)

File: run-checks [hunk 1]
```diff
@@ -32,8 +32,22 @@ case "${1:-all}" in
         exit 1
 esac
 
+CURRENTTRAP="true"
+EXIT_CODE=99
+store_exit_code() {
+    local exit_code=$?
```
Comment (niemeyer) [inline run-checks]: Any reason why this isn't `EXIT_CODE=$?` directly?

**Comment by elopio** [comment]
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA256

On 2015-11-05 09:45, Michael Vogt wrote:
> @elopio <https://github.com/elopio> thanks! I merged with upstream
> and add the "addtrap" for the new trap as well. And no worries,
> shell is a tricky language (despite what people say :)

Yes, it feels like it's time to ask ogra to rewrite the run-checks in
python ^_^.

-----BEGIN PGP SIGNATURE-----
Version: GnuPG v2

iQIcBAEBCAAGBQJWO3txAAoJEIUvYw0No8DgQUAP/2kNM845W8gh2I2hLlnI43DF
JXsQFs8rns1W2RMAGYtc48TtJ8E7+ebQjmc42ydwOzT3O6ICW5iL1frzNfIT0Le5
qALDEbGdCJ5CyFGfzmk7ZTgmzFExqgW3jLI/XBGi+WBgT54eyM84HQEpFOZ0C3i6
oNeIJomAaz/Tu9mQXCb0tdIdY+pl0O+rVDXq5kP7n65orgszLUNzdrbHThkjTXyn
rycT5h3kFIvPOf1+eDrs12KGrEgHpezKQDf93Lmh7yQ7/ooCTMDRSDKw0oHqkeWC
gKF8ZADFzYOswN/cJ/3W+X097AhoDy3n7PV0SbjpWo8LU24OesiOYJmr135lsoya
qD5Qq6k3W94bObn8UDXHJ0vjERqUdITyOpehEnK/aCpshrENwSa44bE1oK01SyHb
BN0prO6+BdXTSbUGo9CySWBFjmTuA4GSVXdJAdy/c4iMFuHdxsssvpqmA6a/eFD1
pSmzWJy2znXH9Yjd3WZTGGJGMGBhqmKcYowx/ertojBk5OBfEMVTJOoXf95FysYF
AckA9AyiTEI7rweXB725SZa3AqdrHpFnCT5gtooUwGTUx0UXwUKTu9X6ccpN12y1
KR8IbpnOr+xZMIUvqYcIgmCcFBO3UYWjqqFSTMZxR/X6hzk9KuSnsosg5M7zi27D
CUQkWPLTsR3EykclM6Ed
=gFq+
-----END PGP SIGNATURE-----

File: run-checks (hunk 1 above)
Comment (mvo5) [inline run-checks]: Indeed, good catch. For some reason I though its needed, but its not, fixed.

**Comment by niemeyer** [comment]
LGTM. Leaving to @chipaca.

**Comment by chipaca** [comment]
The second trap was done in a subshell, so it should have been fine (I tested this when I did that; even added a comment to that effect). In what situation is it not working?

The *third* trap might make the point moot :-)

**Comment by mvo5** [comment]
@chipaca aha, indeed, its the third trap that triggered the issue. We could of course use a subshell here as well if you guys thing this is simpler than the other approach. I don't mind either way.

**Comment by chipaca** [comment]
Eh. This is better than faffing with subshells. :+1:

**Pull request merged**